src/components/application_manager/test/policy_handler_test.cc

```diff
@@ -1467,24 +1668,919 @@ TEST_F(PolicyHandlerTest, SendMessageToSDK) {
+
+  const uint32_t device = 2u;
+
+  EXPECT_CALL(*mock_app_, policy_app_id()).WillOnce(Return(kPolicyAppId_));
+  EXPECT_CALL(app_manager_, connection_handler())
+      .WillRepeatedly(ReturnRef(conn_handler));
+  EXPECT_CALL(conn_handler, get_session_observer())
+      .WillRepeatedly(ReturnRef(mock_session_observer));
+
+  const std::string second_mac_adress = "00:00:00:01:00:00";
+  EXPECT_CALL(mock_session_observer, GetDataOnDeviceID(_, _, NULL, _, _))
+      .WillRepeatedly(
+          DoAll(SetDeviceParamsMacAdress(second_mac_adress), (Return(1u))));
+
+  EXPECT_CALL(mock_message_helper_,
+              SendGetListOfPermissionsResponse(_, kCorrelationKey_, _));
+
+  EXPECT_CALL(*mock_app_, device()).WillRepeatedly(Return(device));
+  EXPECT_CALL(*second_mock_app, device()).WillRepeatedly(Return(device));
+
+  policy_handler_.OnGetListOfPermissions(invalid_connection_key,
+                                         kCorrelationKey_);
+
+  PermissionConsent permissions;
+  permissions.device_id = kDeviceId_;
+  permissions.consent_source = "consent_source";
+
+  policy::FunctionalGroupPermission group_permission_allowed;
+  CreateFunctionalGroupPermission(GroupConsent::kGroupAllowed,
+                                  "allowed",
+                                  "name_allowed",
+                                  group_permission_allowed);
+
+  permissions.group_permissions.push_back(group_permission_allowed);
+
+  EXPECT_CALL(app_manager_, application(_)).Times(0);
+  EXPECT_CALL(app_manager_, application_by_policy_id(_))
+      .WillOnce(Return(second_mock_app))
+      .WillOnce(Return(mock_app_));
+
+  EXPECT_CALL(app_manager_, connection_handler())
+      .WillRepeatedly(ReturnRef(conn_handler));
+  EXPECT_CALL(conn_handler, get_session_observer())
+      .WillRepeatedly(ReturnRef(mock_session_observer));
+
+  EXPECT_CALL(mock_session_observer, GetDataOnDeviceID(_, _, NULL, _, _))
+      .WillOnce(DoAll(SetDeviceParamsMacAdress(kMacAddr_), (Return(1u))))
+      .WillOnce(
+          DoAll(SetDeviceParamsMacAdress(second_mac_adress), (Return(1u))));
+  EXPECT_CALL(*mock_app_, device()).WillRepeatedly(Return(device));
+  EXPECT_CALL(*second_mock_app, device()).WillRepeatedly(Return(device));
+
+  sync_primitives::Lock wait_hmi_lock;
+  sync_primitives::AutoLock auto_lock(wait_hmi_lock);
+  const uint32_t calls_count = 1u;
+  const uint32_t timeout = 1000u;
+  WaitAsync waiter(calls_count, timeout);
+
+  EXPECT_CALL(*mock_policy_manager_, SetUserConsentForApp(_))
+      .WillOnce(NotifyAsync(&waiter));
+
+  policy_handler_.OnAppPermissionConsent(invalid_connection_key, permissions);
+  EXPECT_TRUE(waiter.Wait(auto_lock));
+}
+
+TEST_F(PolicyHandlerTest, GetLockScreenIconUrl_SUCCESS) {
+  EnablePolicyAndPolicyManagerMock();
+  EXPECT_CALL(*mock_policy_manager_, GetLockScreenIconUrl());
+
+  policy_handler_.GetLockScreenIconUrl();
+}
+
+TEST_F(PolicyHandlerTest, RemoveListener_SUCCESS) {
```

Comment:
@okozlovlux, why do we expect there `SUCCESS`? What is condition for that case?
I see plenty of this kind-of names for test cases. Why so?